src/com/xmslyz/luhn/main/Main.java

```diff
@@ -0,0 +1,14 @@
+package com.xmslyz.luhn.main;
+
+import com.xmslyz.luhn.logic.LuhnAlgorithmEngine;
+
+public class Main {
+    public static void main(String[] args) {
+        LuhnAlgorithmEngine luhnAlgorithmEngine = new LuhnAlgorithmEngine();
+        OptionMenu optionMenu = new OptionMenu();
+        String inputNumber = optionMenu.getMainOptions();
+        if (inputNumber != null){
+        OptionMenu.codeNumber(luhnAlgorithmEngine.compute(inputNumber));
```

Comment:
Zerknij na tę linijkę.
```
OptionMenu.codeNumber(luhnAlgorithmEngine.compute(inputNumber));
```
1. Na klasie `OptionMenu` wywołujesz metodę `codeNumber`. Ale parę linijek wyżej [zainstancjonowałeś](https://docs.oracle.com/javase/tutorial/java/javaOO/objectcreation.html) już obiekt `optionMenu`. `codeNumber` nie musi być metodą statyczną: można ją wołać [na obiekcie, a nie klasie](https://www.codementor.io/@stevedonovan/class-vs-object-vs-instance-14i2s2lu6r).  
2. Nazwa metody `codeNumber` jest bardzo myląca. Oczekiwałbym, że po jej wywołaniu dostanę jakiś numerek, a ona nic nie zwraca, tylko wypisuje coś w terminalu. Lepszą nazwą byłoby np. `printChecksum`
3. To jest metoda wypisująca checksumę. Jaki ma ona związek  z `OptionMenu`? Nazwa klasy `OptionMenu` sugeruje, że kod w niej zawarty będzie dotyczył menu opcji wypisywanego użytkownikowi. A wypisanie rezultatu nie wydaje się być tego częścią. Sugeruję przenieść tę metodę do klasy `Main`.